### PR TITLE
Adding Viz Type Picker widget, updating ChartSettings to use styled components

### PR DESCRIPTION
**Pull request by npfitz**
EPIC #26890 

PR to add a new chart type picker, as well as give the rendering layout of the `ChartSettings` component some love. Updated to use styled components where I could, and simplified rendering (IMO) in for Dashcards when we also have previews.

Behavior in Query Builder:
chrome_czq7WZevGT

Layout in Dashcard - One big change is the modal should no longer scroll as a whole. Scrolling should only happen in the settings on the left.
image

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/ChartTypeWidget/ChartTypeWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    } else {
+      return [question.display(), ...suggested.slice(0, 5)];
+    }
+  });
+
+  return (
+    <ChartTypeWidgetRoot data-testid="chart-type-widget">
+      {suggestedVisualizations.map(vizName => (
+        <ChartTypeWidgetIcon
+          key={`${vizName}`}
```
Comment (alxnddr) [inline frontend/src/metabase/query_builder/components/ChartTypeWidget/ChartTypeWidget.tsx:47]: nit key={vizName}

File: frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js [hunk 2]
```diff
@@ -666,15 +667,18 @@ describe("scenarios > visualizations > pivot tables", () => {
...
-      // cy.findByText("Pivot Table")
-      //   .parent()
-      //   .should("have.css", "opacity", "1");
-      cy.icon("pivot_table").click({ force: true });
+    cy.findByText("Settings").click();
+    cy.findByTestId("chart-type-widget").within(() => {
+      cy.icon("chevrondown").click();
     });
 
+    // This part is still failing. Uncomment when fixed.
```
Comment (alxnddr) [inline frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js:675]: Why is this check failing now?

**Review by alxnddr — APPROVED** [review]
Looks & works great 👍

Should we maybe add some unit specs for components and utils?

**Review by gusaiani — APPROVED** [review]
Looking pretty nice overall 👍👍👍

Nothing to add besides what Sasha commented and UX details we're discussing.

File: frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js (hunk 2 above)
Comment (npfitz) [reply frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js:675]: The isSensible check fails when looking at pivot tables. I didn't look into it yet, though it's probably worth noting that I did add an `isSeneisble` check to waterfall charts.

**Comment by npfitz** [comment]
Closing this branch as we're going to take a different approach, and it'll be quicker to start over

**Pull request closed without merging**